### DISCUSSION TITLE
Need bit of help with one particular switch and light power off.

### DISCUSSION THREAD

**Discussion by kriznik** [arendst/Tasmota#21837] [General]
Hello,
for the beginning I want to say thank you for Tasmota, as I'm quite "advanced" user on many many devices since version 4 ... yeah time flies.
Anyway, I'm bit stuck with one (and only this one) device, which acts weirdly, or I'm missing something very obvious somewhere.

I have a wemosD1 which controls mood lights, some bme and dht sensor and PIR sensor connected.

What started to happen is, when light is ON and there is a movement, it goes OFF. So my focus has been to detatch switch from respective power (as I've done on many other devices sucessfully). But does not work.

Can somebody please spot what's wrong with this? Am I using SwitchMode wrong? (previously I had SwitchMode 1, but regarding documentation it might be 3? tried both, was not working..... or what else can be causing this?

```
## CLIMATE Diningtable
	Backlog Module 18; Hostname iotDiningroomTable; Topic home/house/groundfloor/diningroom/table/climate; DeviceName Climate Diningtable; FrinedlyName1 Light Mood; TempOffset 1.3; Fade 1; Speed 7; SwitchMode1 3; SwitchTopic 0;
	{"NAME":"Diningroomtable","GPIO":[0,0,0,0,640,608,0,0,416,1216,160,0,0,0],"FLAG":0,"BASE":18}

	Rule1
			on system#boot      do var1 0 endon
			on switch1#state=1  do if (var1!=1) var1 1; publish home/house/groundfloor/diningroom/motion ON; RuleTimer1 30 else RuleTimer1 30 endif endon
			on Rules#Timer=1    do backlog var1 0; publish home/house/groundfloor/diningroom/motion OFF endon
```

Thanks for hinting me out :)
K.

**Comment by sfromis** [comment]
The basic way of using a PIR is to leave it coupled with the output, no rules or anything. Then you can set a `PulseTime` which will be the minimum time on when movement was detected, and further movement will extend the time. And then you can add rules on top of that, if you really need those publish.

When it comes to the specifics of what you posted, the automatic decoupling when using rules is only when your rules cover all possible state values from the Switch input, and as you have no `Switch1#state=0`, this will have the default action of turning off.

(And you have a typo: `FrinedlyName1`)

**Reply by kriznik** [reply]
well what I want to is that motion (switch) will not interact with power anyhow, because power is controling light which is not related to motion at all. I want simply ignore what it's state, which I thought is SwitchMode1 for ... or do I got it wrongly?

(thanks for typo :))

**Reply by sfromis** [reply]
`SwitchMode 1` will not affect the interaction with local power channels. That's why I mentioned the need to have your rules react to all possible state values (including 0). And you clearly do not want to ignore the state when you have `on switch1#state=1`

You *can* also decouple by forcing MQTT-only with SwitchMode 15/16, but "only" means no rules available. Using `SetOption114` will also decouple.

You could also add a "virtual relay" on an unused gpio, and let the PIR directly control that. Then Tasmota will publish ON/OFF messages automatically, and you can also have your 30 seconds managed there. Hence no rules needed.

**Reply by kriznik** [reply]
what I meant by ignoring state is ignoring state of power. switch1 state is used for PIR, but it's affecting power1 as well. So decoupling is kind of not working for some reason.
Not sure if I understand why this should be handled by rules, I think it's not. I simply want switch not to interfere with power. In other words I dont need/want switch connected to PIR to actually switch anything :)

Strange thing is, that I do have similar devices across the house and all are decoupled just fine, but this one is acting differently 🤷

**Reply by sfromis** [reply]
I already told you why the decoupling did not work, because you have to have rules for both Switch state values, not just one. You could have a rule with `Switch1#state=0` for also handling 0. That way there will be no default action locally. Or one of the other ways of decoupling.

**Reply by kriznik** [reply]
well ... it was not the case. It's now working with SetOption114 and same rules (and same rules I have on other devices which works fine).  Thanks for pointing me to the setoption114 as I was not aware of it while I set it on the previous versions of tasmota with setmode.
